test/e2e/specs/customer_reader.go

```diff
@@ -0,0 +1,88 @@
+package specs
+
+import (
+	"errors"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/test/clients/openshift"
+)
+
+var _ = Describe("Openshift on Azure customer-reader e2e tests [CustomerAdmin]", func() {
+	var (
+		cli       *openshift.Client
+		readercli *openshift.Client
+	)
+
+	BeforeEach(func() {
+		var err error
+		cli, err = openshift.NewSystemAdminClient()
+		if err != nil {
+			Skip(err.Error())
+		}
+		readercli, err = openshift.NewCustomerReaderClient()
+		if err != nil {
+			Skip(err.Error())
+		}
+	})
+
+	It("should not read nodes", func() {
+		_, err := readercli.CoreV1.Nodes().Get("master-000000", metav1.GetOptions{})
+		Expect(kerrors.IsForbidden(err)).To(Equal(true))
+	})
+
+	It("should have read-only on all non-infrastructure namespaces", func() {
+		// create project as enduser
+		namespace, err := randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		Expect(err).ToNot(HaveOccurred())
+		namespace = "e2e-test-" + namespace
+		err = cli.CreateProject(namespace)
```

Comment:
Likewise. This project creation/deletion should be something done by an end user.